sql/com.snowplowanalytics.snowplow/change_form_1.sql

```diff
@@ -1,43 +1,29 @@
---
--- This program is licensed to you under the Apache License Version 2.0,
--- and you may not use this file except in compliance with the Apache License Version 2.0.
--- You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
---
--- Unless required by applicable law or agreed to in writing,
--- software distributed under the Apache License Version 2.0 is distributed on an
--- "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
--- See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
---
--- Authors:       Alex Dean
--- Copyright:     Copyright (c) 2014 Snowplow Analytics Ltd
--- License:       Apache License Version 2.0
--- Release:       2
---
--- Compatibility: iglu:com.snowplowanalytics.snowplow/change_form/jsonschema/1-0-0
+-- AUTO-GENERATED BY igluctl DO NOT EDIT
+-- Generator: igluctl 0.7.0
+-- Generated: 2020-12-11 09:22 UTC
 
-CREATE TABLE atomic.com_snowplowanalytics_snowplow_change_form_1 (
-	-- Schema of this type
-	schema_vendor   varchar(128)  encode runlength not null,
-	schema_name     varchar(128)  encode runlength not null,
-	schema_format   varchar(128)  encode runlength not null,
-	schema_version  varchar(128)  encode runlength not null,
-	-- Parentage of this type
-	root_id         char(36)      encode raw not null,
-	root_tstamp     timestamp     encode raw not null,
-	ref_root        varchar(255)  encode runlength not null,
-	ref_tree        varchar(1500) encode runlength not null,
-	ref_parent      varchar(255)  encode runlength not null,
-	-- Properties of this type
-	form_id         varchar(255)  encode text32k not null,
-	element_id      varchar(255)  encode text32k not null,
-	node_name       varchar(8)    encode runlength not null,
-	type            varchar(15)   encode runlength,
-	element_classes varchar(2048) encode raw, -- Holds a JSON array. TODO: will replace with a ref_ following https://github.com/snowplow/snowplow/issues/647
-	value           varchar(2048) encode raw, -- Cannot enforce not null due to https://github.com/snowplow/snowplow/issues/1134
-	FOREIGN KEY(root_id) REFERENCES atomic.events(event_id)
+CREATE SCHEMA IF NOT EXISTS atomic;
+
+CREATE TABLE IF NOT EXISTS atomic.com_snowplowanalytics_snowplow_change_form_1 (
+    "schema_vendor"   VARCHAR(128)   ENCODE ZSTD NOT NULL,
+    "schema_name"     VARCHAR(128)   ENCODE ZSTD NOT NULL,
+    "schema_format"   VARCHAR(128)   ENCODE ZSTD NOT NULL,
+    "schema_version"  VARCHAR(128)   ENCODE ZSTD NOT NULL,
+    "root_id"         CHAR(36)       ENCODE RAW  NOT NULL,
+    "root_tstamp"     TIMESTAMP      ENCODE ZSTD NOT NULL,
+    "ref_root"        VARCHAR(255)   ENCODE ZSTD NOT NULL,
+    "ref_tree"        VARCHAR(1500)  ENCODE ZSTD NOT NULL,
+    "ref_parent"      VARCHAR(255)   ENCODE ZSTD NOT NULL,
+    "element_id"      VARCHAR(4096)  ENCODE ZSTD NOT NULL,
```

Comment:
I don't think we can merge this (`element_id` and `form_id` are swapped). If user creates the table and uses existing JSONPath (I don't see them in the PR) it means that they will be loading into wrong columns. If we create JSONPath - it means that users who already have tables deployed will be loading into wrong columns. This is the exact the case where patches are super dangerous.

It's okay to have different encodings and even constraints, but order is super crucial here. Especially because almost all clients with `form` schemas still don't have automigrartions.

Reply:
Fixed the order of this and `submit_form`. You learn something new every day 😄 Also ensured they are all consistent with the existing JSONPaths.